components/compositing/compositor.rs

```diff
@@ -1285,30 +1403,52 @@ impl<Window: WindowMethods> IOCompositor<Window> {
 
     fn process_pending_scroll_events(&mut self) {
         let had_events = self.pending_scroll_zoom_events.len() > 0;
-        for event in std_mem::replace(&mut self.pending_scroll_zoom_events,
-                                             Vec::new()) {
-            let delta = event.delta / self.scene.scale;
-            let cursor = event.cursor.as_f32() / self.scene.scale;
 
-            if let Some(ref mut layer) = self.scene.root {
-                layer.handle_scroll_event(delta, cursor);
+        match self.webrender_api {
+            Some(ref webrender_api) => {
+                // Batch up all scroll events into one, or else we'll do way too much painting.
+                let mut total_delta = None;
+                let mut last_cursor = Point2D::zero();
+                for scroll_event in std_mem::replace(&mut self.pending_scroll_zoom_events, vec![]) {
```

Comment:
Can we use `for scroll_event in self.pending_scroll_zoom_events.drain(..) {` now?